suite-native/graph/src/hooks.ts

```diff
@@ -8,14 +8,52 @@ import {
     useGraphForAccounts,
 } from '@suite-common/graph';
 import { selectAccountByKey, selectAccounts } from '@suite-common/wallet-core';
-import { AccountKey } from '@suite-common/wallet-types';
+import { Account, AccountKey } from '@suite-common/wallet-types';
+import { analytics, EventType } from '@suite-native/analytics';
+
+import { timeSwitchItems } from './components/TimeSwitch';
+
+type HoursValue = number | null;
+
+const useWatchTimeframeChangeForAnalytics = (
+    hoursToHistory: HoursValue,
+    account: Account | null = null,
+) => {
+    const isFirstRender = useRef(true);
+
+    useEffect(() => {
+        if (isFirstRender.current) {
+            // Do not report default value on first render.
+            isFirstRender.current = false;
+            return;
+        }
+
+        const timeframeLabel = timeSwitchItems.find(
+            item => item.valueBackInHours === hoursToHistory,
+        )!.label;
```

Comment:
This is unsafe. Could cause an error after some refactoring. Please handle the possibility of undefined in a safe manner. There's a good reason why find method return undefined

Reply:
fixed in 7c3f39874